tests/gnome: enums.h: No such file or directory

```
Failed test: test cases/frameworks/7 gnome
Reason: Compiling source code failed.
The Meson build system
Version: 0.35.0
Source dir: /builddir/build/BUILD/meson-0.35.0/test cases/frameworks/7 gnome
Build dir: /builddir/build/BUILD/meson-0.35.0/b yhmrn5zn
Build type: native build
Project name: gobject-introspection
Native c compiler: cc (gcc 6.2.1)
Build machine cpu family: x86
Build machine cpu: i686
Native dependency gio-2.0 found: YES 2.50.0
Native dependency gio-unix-2.0 found: YES 2.50.0
Native dependency glib-2.0 found: YES 2.50.0
Native dependency gobject-2.0 found: YES 2.50.0
Native dependency gobject-introspection-1.0 found: YES 1.50.0
Native dependency gmodule-2.0 found: YES 2.50.0
Warning, glib compiled dependencies will not work reliably until this upstream issue is fixed: https://bugzilla.gnome.org/show_bug.cgi?id=745754
Program prog.py found: YES (/builddir/build/BUILD/meson-0.35.0/test cases/frameworks/7 gnome/gir/prog.py)
Build targets in project: 23
[1/39] 'glib-compile-resources' '../test cases/frameworks/7 gnome/resources/myresource.gresource.xml' '--sourcedir' '../test cases/frameworks/7 gnome/resources/data' '--sourcedir' '/builddir/build/BUILD/meson-0.35.0/test cases/frameworks/7 gnome/resources' '--c-name' 'myres' '--generate' '--target' 'resources/myresources.c'
[2/39] 'glib-compile-resources' '../test cases/frameworks/7 gnome/resources/myresource.gresource.xml' '--sourcedir' '../test cases/frameworks/7 gnome/resources/data' '--sourcedir' '/builddir/build/BUILD/meson-0.35.0/test cases/frameworks/7 gnome/resources' '--c-name' 'myres' '--generate' '--target' 'resources/myresources.h'
[3/39] cc  '-Igir/girlib@sha' '-fdiagnostics-color=always' '-I../test cases/frameworks/7 gnome/gir' '-Igir' '-pipe' '-Wall' '-Winvalid-pch' '-DMESON_TEST' '-O0' '-g' '-fPIC' '-I/usr/include/glib-2.0' '-I/usr/lib/glib-2.0/include' '-MMD' '-MQ' 'gir/girlib@sha/meson-sample.c.o' '-MF' 'gir/girlib@sha/meson-sample.c.o.d' -o 'gir/girlib@sha/meson-sample.c.o' -c '../test cases/frameworks/7 gnome/gir/meson-sample.c'
[4/39] cc  '-Igir/girprog@exe' '-fdiagnostics-color=always' '-I../test cases/frameworks/7 gnome/gir' '-Igir' '-pipe' '-Wall' '-Winvalid-pch' '-DMESON_TEST' '-O0' '-g' '-I/usr/include/glib-2.0' '-I/usr/lib/glib-2.0/include' '-pthread' '-I/usr/include/gobject-introspection-1.0' '-MMD' '-MQ' 'gir/girprog@exe/prog.c.o' '-MF' 'gir/girprog@exe/prog.c.o.d' -o 'gir/girprog@exe/prog.c.o' -c '../test cases/frameworks/7 gnome/gir/prog.c'
[5/39] cc    -o gir/libgirlib.so 'gir/girlib@sha/meson-sample.c.o' '-Wl,--no-undefined' '-Wl,--as-needed' '-shared' '-fPIC' '-Wl,-soname,libgirlib.so' '-lgobject-2.0' '-lglib-2.0' '-Wl,-rpath,/builddir/build/BUILD/meson-0.35.0/b yhmrn5zn/gir' 
[6/39] 'gdbus-codegen' '--interface-prefix' 'com.example.' '--c-namespace' 'Sample' '--generate-c-code' 'gdbus/generated-gdbus' '../test cases/frameworks/7 gnome/gdbus/com.example.Sample.xml'
[7/39] cc  '-Iresources/resprog@exe' '-fdiagnostics-color=always' '-I../test cases/frameworks/7 gnome/resources' '-Iresources' '-pipe' '-Wall' '-Winvalid-pch' '-DMESON_TEST' '-O0' '-g' '-pthread' '-I/usr/include/glib-2.0' '-I/usr/lib/glib-2.0/include' '-MMD' '-MQ' 'resources/resprog@exe/resources_myresources.c.o' '-MF' 'resources/resprog@exe/resources_myresources.c.o.d' -o 'resources/resprog@exe/resources_myresources.c.o' -c resources/myresources.c
[8/39] cc  '-Iresources/resprog@exe' '-fdiagnostics-color=always' '-I../test cases/frameworks/7 gnome/resources' '-Iresources' '-pipe' '-Wall' '-Winvalid-pch' '-DMESON_TEST' '-O0' '-g' '-pthread' '-I/usr/include/glib-2.0' '-I/usr/lib/glib-2.0/include' '-MMD' '-MQ' 'resources/resprog@exe/main.c.o' '-MF' 'resources/resprog@exe/main.c.o.d' -o 'resources/resprog@exe/main.c.o' -c '../test cases/frameworks/7 gnome/resources/main.c'
[9/39] cc    -o resources/resprog 'resources/resprog@exe/resources_myresources.c.o' 'resources/resprog@exe/main.c.o' '-Wl,--no-undefined' '-Wl,--as-needed' '-lgio-2.0' '-lgobject-2.0' '-lglib-2.0' 
[10/39] "/usr/bin/python3" "/builddir/build/BUILD/meson-0.35.0/meson.py" --internal symbolextractor gir/libgirlib.so 'gir/girlib@sha/gir/libgirlib.so.symbols' 
[11/39] '/usr/bin/python3' '/builddir/build/BUILD/meson-0.35.0/meson.py' '--internal' 'exe' '/builddir/build/BUILD/meson-0.35.0/b yhmrn5zn/meson-private/meson_exe_glib-mkenums_f2544fc7.dat'
[12/39] '/usr/bin/python3' '/builddir/build/BUILD/meson-0.35.0/meson.py' '--internal' 'exe' '/builddir/build/BUILD/meson-0.35.0/b yhmrn5zn/meson-private/meson_exe_glib-mkenums_c6d3c91b.dat'
[13/39] cc    -o gir/girprog 'gir/girprog@exe/prog.c.o' '-Wl,--no-undefined' '-Wl,--as-needed' 'gir/libgirlib.so' '-lglib-2.0' '-lgobject-2.0' '-lglib-2.0' '-lgirepository-1.0' '-lgobject-2.0' '-lglib-2.0' '-Wl,-rpath,/builddir/build/BUILD/meson-0.35.0/b yhmrn5zn/gir' 
[14/39] cc  '-Imkenums/enumprog2@exe' '-fdiagnostics-color=always' '-I../test cases/frameworks/7 gnome/mkenums' '-Imkenums' '-pipe' '-Wall' '-Winvalid-pch' '-DMESON_TEST' '-O0' '-g' '-I/usr/include/glib-2.0' '-I/usr/lib/glib-2.0/include' '-MMD' '-MQ' 'mkenums/enumprog2@exe/mkenums_enums2.c.o' '-MF' 'mkenums/enumprog2@exe/mkenums_enums2.c.o.d' -o 'mkenums/enumprog2@exe/mkenums_enums2.c.o' -c mkenums/enums2.c
FAILED: mkenums/enumprog2@exe/mkenums_enums2.c.o 
cc  '-Imkenums/enumprog2@exe' '-fdiagnostics-color=always' '-I../test cases/frameworks/7 gnome/mkenums' '-Imkenums' '-pipe' '-Wall' '-Winvalid-pch' '-DMESON_TEST' '-O0' '-g' '-I/usr/include/glib-2.0' '-I/usr/lib/glib-2.0/include' '-MMD' '-MQ' 'mkenums/enumprog2@exe/mkenums_enums2.c.o' '-MF' 'mkenums/enumprog2@exe/mkenums_enums2.c.o.d' -o 'mkenums/enumprog2@exe/mkenums_enums2.c.o' -c mkenums/enums2.c
mkenums/enums2.c:5:19: fatal error: enums.h: No such file or directory
 #include "enums.h"
                   ^
compilation terminated.
[15/39] '/usr/bin/python3' '/builddir/build/BUILD/meson-0.35.0/meson.py' '--internal' 'exe' '/builddir/build/BUILD/meson-0.35.0/b yhmrn5zn/meson-private/meson_exe_glib-genmarshal_a2771de5.dat'
[16/39] cc  '-Imkenums/enumprog2@exe' '-fdiagnostics-color=always' '-I../test cases/frameworks/7 gnome/mkenums' '-Imkenums' '-pipe' '-Wall' '-Winvalid-pch' '-DMESON_TEST' '-O0' '-g' '-I/usr/include/glib-2.0' '-I/usr/lib/glib-2.0/include' '-MMD' '-MQ' 'mkenums/enumprog2@exe/main2.c.o' '-MF' 'mkenums/enumprog2@exe/main2.c.o.d' -o 'mkenums/enumprog2@exe/main2.c.o' -c mkenums/main2.c
[17/39] '/usr/bin/python3' '/builddir/build/BUILD/meson-0.35.0/meson.py' '--internal' 'exe' '/builddir/build/BUILD/meson-0.35.0/b yhmrn5zn/meson-private/meson_exe_glib-mkenums_5778956f.dat'
[18/39] '/usr/bin/python3' '/builddir/build/BUILD/meson-0.35.0/meson.py' '--internal' 'exe' '/builddir/build/BUILD/meson-0.35.0/b yhmrn5zn/meson-private/meson_exe_glib-mkenums_26082a12.dat'
[19/39] 'g-ir-scanner' '../test cases/frameworks/7 gnome/gir/meson-sample.c' '../test cases/frameworks/7 gnome/gir/meson-sample.h' '-pthread' '-I/usr/include/gobject-introspection-1.0' '-I/usr/include/glib-2.0' '-I/usr/lib/glib-2.0/include' '--no-libtool' '--namespace=Meson' '--nsversion=1.0' '--warn-all' '--output' 'gir/Meson-1.0.gir' '-I/builddir/build/BUILD/meson-0.35.0/test cases/frameworks/7 gnome/gir' '--include=GObject-2.0' '--cflags-begin' '-DMESON_TEST' '--cflags-end' '--symbol-prefix=meson_' '--identifier-prefix=Meson' '-I/usr/lib/glib-2.0/include' '-I/usr/include/glib-2.0' '-lgobject-2.0' '-lglib-2.0' '-L/builddir/build/BUILD/meson-0.35.0/b yhmrn5zn/gir' '--library' 'girlib'
test cases/frameworks/7 gnome/gir/meson-sample.c:116: Warning: Meson: meson_sample_print_message: invalid return annotation
ninja: build stopped: subcommand failed.
```

Second thing is unrelater, but still would be nice to fix (warning).

--- a/test cases/frameworks/7 gnome/mkenums/meson.build
+++ b/test cases/frameworks/7 gnome/mkenums/meson.build
@@ -23,17 +23,18 @@ test('enum test 1', enumexe1)
 
 # Generate both header and source via template individually and overriding.
 
-enums_c2 = gnome.mkenums('abc2',
-  sources : 'meson-sample.h',
-  c_template : 'enums2.c.in',
-  ftail : '/* trailing source file info */',
-  install_header : true,
-  install_dir : get_option('includedir'))
-
 enums_h2 = gnome.mkenums('abc2',
   sources : 'meson-sample.h',
   h_template : 'enums2.h.in',
   ftail : '/* trailing header file info */',
+  install_header : true,
+  install_dir : get_option('includedir'))
+
+enums_c2 = gnome.mkenums('abc2',
+  sources : 'meson-sample.h',
+  depends : enums_h2,
+  c_template : 'enums2.c.in',
+  ftail : '/* trailing source file info */',
   install_header : true,
   install_dir : get_option('includedir'))
 

